linode/manager
# refactor: [M3-6953] - Refactor Backups Drawer

## Description 📝
- Refactors the Backup Drawer to use modern patterns which will hopefully result in cleaner code and better maintainability 🧹 We're doing this because the Backup Drawer has become a serious pain-point due to its crustyness.

## Major Changes 🔄
- Removes Redux from this component completely 🗑️
  - This component use to manage its state with Redux. This was overkill and added extra complexity. 
- Only mounts this component where it is needed ☑️
  - This component use to be mounded at all times in `MainContent.tsx`. This is overkill because this component is only needed on two pages within the app. It was likely also hurting performance when it was mounted all the time. 
- Lots of refactoring and moving things around to make the code cleaner and more consumable 🧼

## Preview 📷
 ### Error Handling ⛔️
 
 > Shows error handling working as expected

https://github.com/linode/manager/assets/115251059/4097971f-b438-40af-b9ac-6722ea8ed785

 ### Normal Expected Use ✅
 
 > Shows the "happy-path" of how a user might use this

https://github.com/linode/manager/assets/115251059/b2e86a6e-fb4c-4bb2-b893-c2481addc179

## How to test 🧪
- Verify the Backups drawer works as expected
  - Behavior and UX should be mostly the same as before but please verify there are no changes that negatively impacted the UX
  - Verify you can enable backups
  - Verify you can enable backup auto-enrollment
  - Consider blocking a network request to verify the error state looks good
  
### How to make this component show up ❓
- Go to `http://localhost:3000/profile/settings?preferenceEditor=true` and set your preferences to `{}` and **Save**
- Deploy a Linode with backups disabled
- Make sure you are on an unrestricted and non-managed account
- You should see the Backups "Call to Action" banner on the Linodes Landing page 🎉
